docs: update usage doc with info about `--watch` flag and absence of `--version` flag (#1142)
asyncapi/generator: docs/usage.md

- [Generator library](#using-as-a-modulepackage)

## AsyncAPI CLI
Generates whatever you want using templates compatible with AsyncAPI Generator.
```bash
USAGE
  $ asyncapi generate fromTemplate [ASYNCAPI] [TEMPLATE] [-h] [-d <value>] [-i] [--debug] [-n <value>] [-o <value>] [--force-write] [-w] [-p <value>] [--map-base-url <value>]

ARGUMENTS

>   ASYNCAPI  - Local path, url or context-name pointing to AsyncAPI file

Reviewer: ```suggestion
  ASYNCAPI  - Local path, URL pointing to AsyncAPI document  for example https://bit.ly/asyncapi
```

Author: Hi @Florence-Njeri , @derberg 
I did "construct" my update by copy/pasting output of `asyncapi generate fromTemplate --help`. The suggestions break alignment between code and documentation. Do you really want to go that direction?

Alternative: apply suggestions to code (help text) and documentation in parallel. @derberg, can you point me to the help text?

Reviewer 2: @Florence-Njeri normally I like that you want to clarify help with examples, like example link. Problem is that https://bit.ly/asyncapi is no longer the best as it points to asyncapi `2.0.0` and nobody knows who owns this short link and how to update it. So we should not promote it

so correct help is
`Local path, name of CLI context or URL pointing to AsyncAPI document`

Reviewer 2: @eelcofolkertsma we can accept the update here now and if you could followup with PR to fix in in CLI in https://github.com/asyncapi/cli/blob/master/src/commands/generate/fromTemplate.ts#L106 that would be awesome

  TEMPLATE  - Name of the generator template like for example @asyncapi/html-template or https://github.com/asyncapi/html-template

> 

Reviewer: @derberg correct me if I am wrong but I feel the changes being made to this file might be unnecessary especially this part `asyncapi generate fromTemplate [ASYNCAPI] [TEMPLATE] [-h] [-d <value>] [-i] [--debug] [-n <value>] [-o <value>] [--force-write] [-w] [-p <value>] [--map-base-url <value>]` why not just leave it as was `asyncapi generate fromTemplate <asyncapi> <template> [<options>]`

Author: See earlier argument. Keep documentation in sync with help-output generated from code

Reviewer 2: yeah, I think in this particular case good to follow the output from CLI that is following best practices

}
```

> See the [API documentation](/docs/tools/generator/api) for more examples and full API reference information.

Reviewer: This leads to a 404 error. We should leave it as is @derberg any thought about this?

Author: I do not know how/why link was incorrect, but I have accepted suggestion so now reference is working again

Reviewer 2: @Florence-Njeri actually both are good - not sure where you see 404

I personally prefer full relative links like `/docs/tools/generator/api` but `api` works as well. Let's leave it as `api` as my approach for full relative links is an old habit that I follow but brainfart every time I try to find in my brain the reason for doing it 😄

> See the [API documentation](/docs/tools/generator/api) for more examples and full API reference information.

Reviewer: ```suggestion
See the [API documentation](api) for more examples and full API reference information.
```
